kvbc/src/reconfiguration_kvbc_handler.cpp

```diff
@@ -336,41 +357,61 @@ bool ReconfigurationHandler::handle(const messages::UnwedgeCommand& cmd,
                                     concord::messages::ReconfigurationResponse&) {
   if (!bftEngine::ControlStateManager::instance().getCheckpointToStopAt().has_value()) {
     LOG_INFO(getLogger(), "replica is already unwedge");
+    return true;
   }
   LOG_INFO(getLogger(), "Unwedge command started " << KVLOG(cmd.bft_support));
-  auto& controlStateManager = bftEngine::ControlStateManager::instance();
-  bool valid = controlStateManager.verifyUnwedgeSignatures(cmd.signatures, cmd.bft_support);
-  if (valid) {
-    auto newEpoch = bftEngine::EpochManager::instance().getSelfEpochNumber() + 1;
-    concord::kvbc::categorization::VersionedUpdates ver_updates;
-    ver_updates.addUpdate(std::string{kvbc::keyTypes::reconfiguration_epoch_key},
-                          concordUtils::toBigEndianStringBuffer(newEpoch));
-    auto block_id = persistReconfigurationBlock(ver_updates, bft_seq_num);
-    bftEngine::EpochManager::instance().setSelfEpochNumber(newEpoch);
-    bftEngine::EpochManager::instance().setGlobalEpochNumber(newEpoch);
-    LOG_INFO(getLogger(), "Starting new epoch " << KVLOG(newEpoch, block_id));
-    controlStateManager.setStopAtNextCheckpoint(0);
-    bftEngine::IControlHandler::instance()->resetState();
-    LOG_INFO(getLogger(), "Unwedge command completed sucessfully");
+  auto curr_epoch = bftEngine::EpochManager::instance().getSelfEpochNumber();
```

Comment:
refactoring + change the unwedge signature to contain the replica's epoch number